assets/training/model_management/src/azureml/model/mgmt/processors/transformers/convertors.py

```diff
@@ -175,7 +177,65 @@ def _save(
             else:
                 logger.info(f"Successully installed {pkgs}. pip logs =>\n{stdout}\n")
 
-        if segregate:
+        # Set experimental flag
+        if self._experimental:
+            logger.info("Experimental features enabled for MLflow conversion")
+            self._hf_conf["exp"] = True
+
+        try:
+            # create a conda environment for OSS transformers Flavor
+            python_version = platform.python_version()
+            pip_pkgs = self._get_curated_environment_pip_package_list()
+            conda_deps = CondaDependencies.create(conda_packages=None,
+                                                  python_version=python_version,
+                                                  pip_packages=pip_pkgs,
+                                                  pin_sdk_version=False)
+
+            curated_conda_env = conda_deps.as_dict()
+
+            # handle OSS trust_remote_code value
+            trust_remote_code_val = False
+            config_args = self._hf_conf.get(HF_CONF.HF_CONFIG_ARGS.value, {})
+            tokenizer_args = self._hf_conf.get(HF_CONF.HF_TOKENIZER_ARGS.value, {})
+            model_args = self._hf_conf.get(HF_CONF.HF_MODEL_ARGS.value, {})
+
+            if (config_args.get('trust_remote_code', False)
+                    and tokenizer_args.get('trust_remote_code', False)
+                    and model_args.get('trust_remote_code', False)):
+                trust_remote_code_val = True
+
+            # handle OSS model loading with device map
+            try:
+                model_pipeline = transformers.pipeline(task=self._task, model=model, device_map="auto",
+                                                       trust_remote_code=trust_remote_code_val)
+                logger.info("OSS model loaded with device_map auto")
+            except Exception as e:
+                logger.error("OSS Model load failed with exception: {}, reloading without device_map auto".format(e))
+                model_pipeline = transformers.pipeline(task=self._task, model=model, trust_remote_code=trust_remote_code_val)
+
+                # pass in signature for a text-classification model
+            if self._task == SupportedNLPTasks.TEXT_CLASSIFICATION.value:
+                inputs = Schema([ColSpec(DataType.string)])
+                outputs = None
+                params = ParamSchema([ParamSpec("return_all_scores", "boolean", default=True)])
+                self._signatures = ModelSignature(inputs=inputs, outputs=outputs, params=params)
```

Comment:
handle in derived class?

Reply:
Lets move to specific params changes within specific task classes. In this case: NLPMLflowConvertor

Reply:
Did that for text-gen, text classification is automated